pulsar-broker-common/src/main/java/org/apache/pulsar/broker/authentication/AuthenticationDataSource.java

```diff
@@ -153,4 +153,18 @@ default String getSubscription() {
      * Subscription name can be necessary for consumption.
      */
     default void setSubscription(String subscription) { };
+
+    /**
+     * Set a property.
+     * This property can be used post-authn or post-authz.
+     */
+    default void setProperty(String key, Object value) { };
+
+    /**
+     * Get a property.
+     * @return the match property by the key.
+     */
+    default Object getProperty(String key) {
+        return null;
+    }
```

Comment:
These methods are new, so maybe

1. Throwing an exception like `UnsupportedOperationException`
2. Add `@since` to `JavaDoc`

Reply:
> Throwing an exception like UnsupportedOperationException
Should keep the same with other methods.

Reply:
True,